Makefile

```diff
@@ -1,28 +1,91 @@
+DEPS := "wget git go docker"
 
-all: build_local
+BINARY := autospotting
+BINARY_PKG := ./core
+COVER_PROFILE := /tmp/coverage.out
+BUCKET_NAME ?= cloudprowess
+LOCAL_PATH := build/s3/dv
 
-clean:
-	./build_helper.sh clean
+DOCKER_IMG := eawsy/aws-lambda-go
 
-prepare_bindata:
-	./build_helper.sh prepare_bindata
+GO_FILES := $(shell find ./ -name '*.go')
 
-build_lambda_binary: prepare_bindata
-	./build_helper.sh build_lambda_binary
+BINDATA_DIR := data
+BINDATA_FILE := generated_bindata.go
 
-prepare_upload_data: build_lambda_binary
-	./build_helper.sh prepare_upload_data
+SHA := $(shell git rev-parse HEAD | cut -c 1-7)
+BUILD := $(or $(TRAVIS_BUILD_NUMBER), $(TRAVIS_BUILD_NUMBER), $(SHA))
 
-# the following targets are only used for local development
+# upstream data
+EC2_INSTANCES_INFO_COMMIT_SHA := e655e36660bc617713c4ef9a1409cc65a209cb27
+INSTANCES_URL := "https://raw.githubusercontent.com/powdahound/ec2instances.info/$(EC2_INSTANCES_INFO_COMMIT_SHA)/www/instances.json"
 
-build_local: prepare_bindata
-	./build_helper.sh build_local
+all: build_local                                             ## Build the code
 
-upload: prepare_upload_data
-	./build_helper.sh upload
+clean:                                                       ## Remove installed packages/temporary files
+	go clean ./...
+	rm -rf $(BINDATA_DIR) $(BINDATA_FILE)
+	docker rmi -f $(DOCKER_IMG) 2> /dev/null || true
 
-test: build_local
-	./autospotting
+check_deps:                                                  ## Verify the system has all dependencies installed
+	@for DEP in $(shell echo "$(DEPS)"); do \
+		command -v "$$DEP" &>/dev/null \
+		|| (echo "Error: dependency '$$DEP' is absent" ; exit 1); \
+	done
+	@echo "all dependencies satisifed: $(DEPS)"
 
-cover:
-	./build_helper.sh calculate_coverage
\ No newline at end of file
+bindata: check_deps                                          ## Convert instance data into go file
+	@type go-bindata || go get -u github.com/jteeuwen/go-bindata/...
+	@mkdir -p $(BINDATA_DIR)
+	wget --quiet -nv -c $(INSTANCES_URL) -O $(BINDATA_DIR)/instances.json
+	@echo $(BUILD) > $(BINDATA_DIR)/BUILD
+	go-bindata -o $(BINDATA_FILE) -nometadata $(BINDATA_DIR)
+	@gofmt -w $(BINDATA_FILE)
+
+build_lambda_binary: bindata                                 ## Build lambda binary
+	docker run --rm -v $(GOPATH):/go -v $(PWD):/tmp $(DOCKER_IMG)
+
+prepare_upload_data: build_lambda_binary                     ## Create archive to be uploaded
+	@rm -rf $(LOCAL_PATH)
+	@mkdir -p $(LOCAL_PATH)
+	@mv handler.zip $(LOCAL_PATH)/lambda.zip
+	@cp -f cloudformation/stacks/AutoSpotting/template.json $(LOCAL_PATH)/template.json
+	@cp -f cloudformation/stacks/AutoSpotting/template.json $(LOCAL_PATH)/template_build_$(BUILD).json
+	@cp -f $(LOCAL_PATH)/lambda.zip $(LOCAL_PATH)/lambda_build_$(BUILD).zip
+	@cp -f $(LOCAL_PATH)/lambda.zip $(LOCAL_PATH)/lambda_build_$(SHA).zip
```

Comment:
Oh okay, sorry for the misunderstanding, then I'm fine with either way. As you said build number might be easier/more readable, we can drop the SHA and see later if it doesn't cause problem. But might be better to do that in another PR I believe.